src/python/zquantum/core/graph.py

```diff
@@ -73,163 +66,223 @@ def generate_graph_node_dict(graph: nx.Graph) -> dict:
     return nodes_dict
 
 
+Sampler = Iterator[float]
+
+
+def uniform_sampler(min_value=0, max_value=1) -> Sampler:
+    while True:
+        yield random.uniform(min_value, max_value)
+
+
+def constant_sampler(value) -> Sampler:
+    return itertools.repeat(value)
+
+
+def static_sampler() -> Sampler:
+    return constant_sampler(1.0)
+
+
+def choice_sampler(choices) -> Sampler:
+    while True:
+        yield random.choice(choices)
+
+
+def uniform_range_sampler(*range_args) -> Sampler:
+    while True:
+        choices = range(*range_args)
+        yield random.choice(choices)
```

Comment:
Additionally, I don't find this one too intuitive. I would probably just use `choice_sampler` myself, but I'm not against keeping this if you think it has value.

Reply:
Maybe it was included in Fred's code for some reason?

Reply:
Yeah, it wasn't for me either, I'll remove it